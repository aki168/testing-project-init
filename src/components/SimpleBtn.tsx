import { useState } from "react";

export const SimpleBtn: React.FC = () => {
  const [state, setState] = useState(false);

  return (
    <button
      onClick={() => {
        setState((prev) => !prev);
      }}
    >
      {state ? "ON" : "OFF"}
    </button>
  );
};
