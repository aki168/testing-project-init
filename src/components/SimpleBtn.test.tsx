import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SimpleBtn } from "./SimpleBtn";

test("click and show ON or OFF", () => {
  const view = render(<SimpleBtn />);
  expect(view.container).toMatchSnapshot();
  // const btn = screen.getByRole("button");
  // expect(btn).toHaveTextContent("OFF");
  // userEvent.click(btn);
  // expect(btn).toHaveTextContent("ON");
});
