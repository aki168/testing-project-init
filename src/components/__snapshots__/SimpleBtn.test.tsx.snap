// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click and show ON or OFF 1`] = `
<div>
  <button>
    OFF
  </button>
</div>
`;
